apps/pipelines/models.py

```diff
@@ -167,11 +167,16 @@ def node_ids(self):
 
     def simple_invoke(self, input: str) -> PipelineState:
         """Invoke the pipeline without a session or the ability to save the run to history"""
-
         from apps.pipelines.graph import PipelineGraph
+        from apps.utils.factories.experiment import ExperimentSessionFactory
 
-        runnable = PipelineGraph.build_runnable_from_pipeline(self)
-        output = runnable.invoke(PipelineState(messages=[input]))
+        output = ""
+        with transaction.atomic():
+            runnable = PipelineGraph.build_runnable_from_pipeline(self)
+            session = ExperimentSessionFactory()
```

Comment:
Cool, I'm onboard with this

Reply:
Thoughts on this?https://github.com/dimagi/open-chat-studio/pull/950/commits/6cb36c6adc75a803265f3c0ef157191fbacb0621.

it's either this approach, or we have to have some "dry run" check where it makes sense to prevent db queries